Cython/Compiler/Optimize.py

```diff
@@ -339,20 +339,22 @@ def _transform_reversed_iteration(self, node, reversed_function):
             PyrexTypes.CFuncTypeArg("s", Builtin.bytes_type, None)
             ])
 
-    def _transform_indexable_iteration(self, node, slice_node, reversed=False):
+    def _transform_indexable_iteration(self, node, slice_node, is_mutable, reversed=False):
         """In principle can handle any iterable that Cython has a len() for and knows how to index"""
         unpack_temp_node = UtilNodes.LetRefNode(
             slice_node.as_none_safe_node("'NoneType' is not iterable"),
-            may_hold_none=False
+            may_hold_none=False, is_temp=True
             )
 
         start_node = ExprNodes.IntNode(
             node.pos, value='0', constant_result=0, type=PyrexTypes.c_py_ssize_t_type)
-        length_temp = UtilNodes.LetRefNode(ExprNodes.SimpleCallNode(node.pos,
+        length_call = ExprNodes.SimpleCallNode(node.pos,
                                     function = ExprNodes.NameNode(node.pos, name="len"),
```

Comment:
I wonder what happens if users override `len` in their code with something else. We should probably make sure that this is identified as the builtin.